### ISSUE TITLE
continuity SFH forms stars past its last bin edge, so a correctly bounded ladder still warns about the Big Bang

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#1978]
Split out from #1975, which fixed the plumbing. This is the behavior that remains once redshift-matched edges actually reach the forward pass.

## The clip

`continuity` assigns each age to a bin and then clamps the index:

```python
bin_idx = jnp.searchsorted(bin_edges_yr, age_yr, side="right") - 1
bin_idx = jnp.clip(bin_idx, 0, n_bins - 1)
sfr = sfr_bins[bin_idx]
```

Every age older than the last edge is clamped into the oldest bin, so the SFH keeps forming stars at that bin's rate all the way to the end of the age grid (13.804 Gyr), outside any declared bin. The mass normalization sums bin widths only, so this tail is not normalized against anything.

With the default ladder ending at 13.7 Gyr this was invisible. It becomes visible the moment a ladder is bounded to the age of the universe, which is exactly what #1975 made possible.

## Symptom: a false Big-Bang warning on a correctly bounded model

Measured at z = 1.5 (universe 4.28 Gyr), `log_total_mass = 10.30`, 8 edges log-spaced to the age of the universe:

| ladder | flat SFH `log_mstar_formed` | rising SFH | Big-Bang warning |
| --- | --- | --- | --- |
| default, to 13.7 Gyr | 9.796 | 10.205 | 66% / 19% |
| z-matched, to 4.28 Gyr | 10.301 | 10.299 | 66% / 21% |

The z-matched ladder normalizes correctly (10.301 against a declared 10.30), yet still reports that 66% of the mass formed before the Big Bang, because the warning integrates the clipped tail. A user who bounds their bins correctly, which is the documented fix for this warning, gets the same warning as before and no way to silence it honestly.

## Suggested fix

Zero the SFR above the last edge rather than clamping into the oldest bin:

```python
sfr = jnp.where(age_yr > bin_edges_yr[-1], 0.0, sfr)
```

Leave the low-end clamp alone: `psb_suess2022` uses edges starting at 0.3 Gyr and relies on ages below the first edge taking the youngest bin's rate.

This affects the three models whose shape function is `continuity` (`continuity`, `bursty_continuity`, `prospector_beta`). For the default ladder the change touches only the 13.7 to 13.804 Gyr sliver.

## Related: the notebooks hide this warning globally

`notebooks/_setup.py::quiet()` filters `".*before the Big Bang.*"` for every notebook, alongside genuinely cosmetic notices like the baked-in nebular one. With the default ladder at z > 0 that warning is not cosmetic: it is a half-dex error in the mass parameter. Worth reconsidering once the clip is fixed and the warning only fires when something is actually wrong.

## Acceptance criteria

- [ ] A `continuity` SFH whose last edge is the age of the universe at the fit redshift emits no Big-Bang warning.
- [ ] `sfr_history` is exactly zero for lookback times beyond the last supplied edge.
- [ ] `psb_suess2022` behavior below its first edge is unchanged.
- [ ] Default-ladder photometry changes by no more than the 13.7 to 13.804 Gyr sliver.
